src/beatmap/beatmap.h

```diff
@@ -478,6 +478,7 @@ struct oshu_metadata {
 	 * May be *NULL*.
 	 */
 	char **tags;
+	char *_tags;
```

Comment:
No way. The tags string is something we allocate ourselves, so we can easily guarantee that `tags[0] == _tags`.
How the tags array should be freed should be mentionned in the documentation of the field.

Reply:
lol, we just talked about this, yesterday.